src/Tizen.NUI/src/internal/Common/WeakEvent.cs

```diff
@@ -106,13 +124,46 @@ public WeakHandler(U handler)
                 methodInfo = d.Method;
             }
 
+            public bool Equals(object? otherTarget, MethodInfo otherMethodInfo)
+            {
+                return otherTarget == weakReference?.Target && otherMethodInfo.Equals(methodInfo);
+            }
+
             public bool Equals(U handler)
             {
                 Delegate other = (Delegate)(object)handler;
-                return other != null && other.Target == weakReference?.Target && other.Method.Equals(methodInfo);
+                var rooting = other?.Target;
+                return other != null && Equals(rooting, other.Method);
             }
 
-            public bool IsAlive => weakReference == null || weakReference.IsAlive;
+            public bool Equals(WeakHandler<U>? other)
+            {
+                var rooting = other?.weakReference?.Target;
+                return other != null && Equals(rooting, other.methodInfo);
+            }
+
+            public override int GetHashCode()
+            {
+                return weakReference.Target.GetHashCode() ^ methodInfo.GetHashCode();
+            }
+
+            public bool IsAlive
+            {
+                get
+                {
+                    var rooting = weakReference?.Target;
+
+                    if (weakReference == null) return true;
```

Comment:
```suggestion
                    if (weakReference == null) return true; /// Static handler case. We don't check the target lifecycle for this case.
```

Could you please add some comment why we return `true` in null case?

Reply:
Good suggestion. I created a private method `IsStatic` to increase code readablity.